src/components/alert/alert.twig

```diff
@@ -1,7 +1,19 @@
 {% set tag_name = tag_name|default('div') %}
 
-<{{ tag_name }} class="c-alert{% if class %} {{ class }}{% endif %}">
+<{{ tag_name }} class="c-alert{% if floating %} c-alert--floating{% endif %}{% if class %} {{ class }}{% endif %}">
   <div class="c-alert__inner">
+    {% if block('icon')|trim or icon %}
+      <span class="c-alert__extra">
+        {% block icon %}
+          {% if icon %}
+            {% include '@cloudfour/components/icon/icon.twig' with {
+              name: icon,
+              aria_hidden: true
+            } only %}
+          {% endif %}
+        {% endblock %}
+      </span>
+    {% endif %}
```

Comment:
In a separate PR I've been finding a way to simplify checks like this from what's being done in patterns like our button.

Earlier in the template, we can define our block but capture its output using `set`:

```twig
{% set _icon_block %}
  {% block icon %}
    {% if icon %}
      {% include '@cloudfour/components/icon/icon.twig' with {
        name: icon,
        aria_hidden: true
      } only %}
    {% endif %}
  {% endblock %}
{% endset %}
```

Then later in the template we can do this:

```twig
{% if _icon_block|trim is not empty %}
  <span class="c-alert__extra">
    {{_icon_block}}
  </span>
{% endif %}
```

Slightly less code, and to my eye makes the template _slightly_ easier to understand. But, that may be a matter of opinion... push back if you disagree!